docs-tech-guide.rst

```diff
@@ -0,0 +1,557 @@
+
+.. tip::
+  - The ``clone`` command creates a new directory inside the current one. So you do not need to create a new `odk-docs` directory first.
+  - As noted above, we recommend a master :file:`odk` directory that holds your virtualenv directory and your git repo in two separate subdirectories. So you would be in that master :file:`odk` directory when you clone down the repo.
+  - Double check that right folders are in the right places
+
+  .. code-block:: none
+
+    - odk/
+      - odkenv/
+      - docs/
+
+.. _upstream-the-docs:
+
+Set the Upstream Remote
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you clone down a repo, the local copy calls your GitHub copy ``origin``. You should also set ``upstream`` as the name of the original, main GitHub repo.
+
+
+.. code-block:: console
+
+  $ git remote add upstream https://github.com/opendatakit/docs.git
+
+Or in Windows:
+
+.. code-block:: doscon
+
+  > git remote add upstream https://github.com/opendatakit/docs.git
+
+Run ``git remote -v`` to check the status, you should see something like this:
+
+.. code-block:: console
+
+  $ origin https://github.com/your-github-username/docs.git (fetch)
+  $ origin https://github.com/your-github-username/docs.git (push)
+  $ upstream https://github.com/opendatakit/docs.git (fetch)
+  $ upstream https://github.com/opendatakit/docs.git (push)
+
+.. _install-doc-dependencies:
+
+Install Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The first time you clone down the repo, you'll need to install the dependencies. Make sure you have your Python 3 virtual environment set up and activated in the docs repo and then:
+
+.. code-block:: console
+
+  $ pip install -r requirements.txt
+
+.. note::
+
+  If you are working on the design, testing, or deployment of the docs, you might find the need to install an additional PyPi package. If you do, please update the requirements.txt file with ``pip freeze > requirements.txt``. Pull Requests which change :file:`requirements.txt` should include a note about why the new packages are needed.
+
+.. note::
+
+  If you have problems when running the Sphinx commands (see below), you may have a dependency issue. Try running ``pip install -r requirements.txt`` again.
+
+.. _docs-workflow-details:
+
+Workflow Details
+-------------------
+
+.. _git-pull-the-docs:
+
+Pull in Updates from Upstream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You probably won't need to do this the first time, but you should always pull in any changes from the main repository before working.
+
+
+.. code-block:: console
+
+  $ git pull upstream
+
+.. note::
+
+  If you get this message:
+
+  .. code-block:: none
+
+        You asked to pull from the remote 'upstream', but did not specify a branch.
+        Because this is not the default configured remote for your current branch,
+        you must specify a branch on the command line.
+
+
+  Try running ``git pull upstream master`` instead.
```

Comment:
A note can be added here to checkout to master branch before pulling the upstream content.